meson.build

```diff
@@ -0,0 +1,266 @@
+    lib_srcs,
+    include_directories: incl_dirs,
+    dependencies: lib_deps,
+    install: true)
+endif
+
+if enable_shared
+  if enable_shared_version
+    shared_link_args = '-Wl,--version-script=@0@/libisns.vers'.format(meson.current_source_dir())
+  else
+    shared_link_args = []
+  endif
+  libisns_lib = shared_library('isns',
+    lib_srcs,
+    include_directories: incl_dirs,
+    dependencies: lib_deps,
+    link_args: shared_link_args,
+    version: '0',
+    install: true)
+endif
+
+libisns_dep = declare_dependency(link_with: libisns_lib)
+
+#
+# build binaries
+#
+
+our_dependencies = [config_dep, security_dep, libisns_dep]
+if with_slp
+  our_dependencies += cc.find_library('slp')
+endif
+
+executable('isnsd',
+  sources: files('isnsd.c'),
+  include_directories: incl_dirs,
+  dependencies: our_dependencies,
+  install: true,
+  install_dir: sbin_dir)
+
+executable('isnsdd',
+  sources: files(['isnsdd.c', 'local.c']),
+  include_directories: incl_dirs,
+  dependencies: our_dependencies,
+  install: true,
+  install_dir: sbin_dir)
+
+executable('isnsadm',
+  sources: files('isnsadm.c'),
+  include_directories: incl_dirs,
+  dependencies: our_dependencies,
+  install: true,
+  install_dir: sbin_dir)
+
+# a testing program (not installed, not currently used here)
+e = executable('bitvector',
+  sources: files('bitvector.c'),
+  include_directories: incl_dirs,
+  dependencies: our_dependencies,
+  c_args: ['-DTEST'])
+test('test bitvector', e)
+
+# build our test files ??? (FIXME: not working)
+pyunit_test_works = false
+if pyunit_test_works
+  test_progs = []
+  foreach t: tests
+    e = executable(t,
+      sources: files('tests' / t + '.c'),
+      include_directories: incl_dirs,
+      dependencies: our_dependencies)
+    test_progs += e
+  endforeach
+endif
+
+#
+# man pages
+#
+
+foreach m: man8_pages
+  install_man(m, install_dir: man_dir / 'man8')
+endforeach
+foreach m: man5_pages
+  install_man(m, install_dir: man_dir / 'man5')
+endforeach
+
+#
+# config files
+#
+
+foreach c: config_files
+  install_data(c,
+    install_dir: etc_dir / 'isns')
+endforeach
+
+#
+# systemd service files
+#
+
+systemd_files = files([
+  'isnsd.service',
+  'isnsd.socket'])
+foreach f: systemd_files
+  install_data(f,
+    install_dir: systemd_unit_dir)
+endforeach
+
+#
+# our package file
+#
+
+pkg_mod = import('pkgconfig')
+pkg_mod.generate(libraries: libisns_lib,
+  version: '0',
+  name: 'libisns',
+  description: 'iSNS userspace library')
+
+#
+# public header files
+#
+
+# one template file, and the rest are static
+if false
+  configure_file(
+    input: template_hdr_file,
+    output: template_hdr_name,
+    configuration: conf,
+    install: true,
+    install_dir: get_option('includedir') / 'libisns')
+  install_headers(static_hdr_files,
+    install_dir: get_option('includedir') / 'libisns')
+endif
+
+#
+# create our runtime directory (empty)
+#
+
+install_subdir('isns',
+  install_dir: '/var/lib',
+  install_mode: 'r--------')
```

Comment:
I would leave a comment here:

```
# this can be replaced with install_emptydir once the minimum requirement of Meson is bumped to >=0.60.0
```

Note that the use of install_subdir() is called out in the docs as "possible to use in this manner due to a clever loophole, but doing so is deprecated".

Of course, deprecated functionality is perfectly reasonable to use if you're still supporting older systems!

...

I'm not sure if the install_subdir hack supports install modes, I *think* that was one of the reasons that made me create install_emptydir instead.

Reply:
fixed